### ISSUE TITLE
Duplicates time when fill(0)

### ISSUE THREAD

**Issue by NamPNQ** [influxdb/influxdb#1190]
I get problem when get data by group by time and fill (0)

Here is my data:
```
"GET /db/stg:filestring/series?q=select+count%28timespan%29+as+visits+from+%22tracking_file%22+where+sharer_id+%3D+%2734f72b47-847c-4a12-a64e-43f459d6e9aa%27+and+file_id+%3D+%27673b65b4-0866-4020-a391-58697d0e2422%27++and+time+%3E+1404207480s+and+time+%3C+1417513080s++group+by+time%281d%29+fill%280%29+order+asc&p=root&chunked=false&time_precision=ms&u=root HTTP/1.1"
```
```   
 [{
    u 'points': [
        [1416209245, 245010001, u '34f72b47-847c-4a12-a64e-43f459d6e9aa', 6, u 'Web', u '20.972500', u '34f72b47-847c-4a12-a64e-43f459d6e9aa', u '673b65b4-0866-4020-a391-58697d0e2422', u 'VN', u '105.777000', u 'd165d6a2-a063-4ff0-bb02-07f30923f99d'],
        [1416196764, 245810001, u '34f72b47-847c-4a12-a64e-43f459d6e9aa', 24, u 'Web', u '20.972500', u '34f72b47-847c-4a12-a64e-43f459d6e9aa', u '673b65b4-0866-4020-a391-58697d0e2422', u 'VN', u '105.777000', u 'd165d6a2-a063-4ff0-bb02-07f30923f99d'],
        [1405137578, 371160001, u '34f72b47-847c-4a12-a64e-43f459d6e9aa', 19, u 'Web', u '10.750000', u '34f72b47-847c-4a12-a64e-43f459d6e9aa', u '673b65b4-0866-4020-a391-58697d0e2422', u 'VN', u '106.667000', u 'd2314169-41bc-4919-a70b-9cda24f2e2db'],
        [1405137449, 371180001, u '34f72b47-847c-4a12-a64e-43f459d6e9aa', 102, u 'Web', u '10.750000', u '34f72b47-847c-4a12-a64e-43f459d6e9aa', u '673b65b4-0866-4020-a391-58697d0e2422', u 'VN', u '106.667000', u 'd2314169-41bc-4919-a70b-9cda24f2e2db'],
        [1403253505, 458260001, u '34f72b47-847c-4a12-a64e-43f459d6e9aa', 137, u 'Web', u '10.750000', u '34f72b47-847c-4a12-a64e-43f459d6e9aa', u '673b65b4-0866-4020-a391-58697d0e2422', u 'VN', u '106.667000', u 'af84f83e-cdaa-45aa-a9ae-db22015f2051']
    ], u 'name': u 'tracking_file', u 'columns': [u 'time', u 'sequence_number', u 'sharer_id', u 'timespan', u 'platform', u 'lat', u 'owner_id', u 'file_id', u 'country_code', u 'long', u 'viewer_id']
}]
```

```python
from influxdb import InfluxDBClient
client = InfluxDBClient()
data = client.query("select count(timespan) as visits from tracking_file where sharer_id = '34f72b47-847c-4a12-a64e-43f459d6e9aa' and file_id = '673b65b4-0866-4020-a391-58697d0e2422'  and time > 1403253505s and time < 1417514972s  group by time(1d) fill(0) order asc")
for p in data[0]['points']:
    print datetime.fromtimestamp(int(p[0]))
```

I get result 
```
2014-06-20 00:00:00
2014-06-21 00:00:00
2014-06-22 00:00:00
2014-06-23 00:00:00
2014-06-24 00:00:00
2014-06-25 00:00:00
2014-06-26 00:00:00
2014-06-27 00:00:00
2014-06-28 00:00:00
2014-06-29 00:00:00
2014-06-30 00:00:00
2014-07-01 00:00:00
2014-07-02 00:00:00
2014-07-03 00:00:00
2014-07-04 00:00:00
2014-07-05 00:00:00
2014-07-06 00:00:00
2014-07-07 00:00:00
2014-07-08 00:00:00
2014-07-09 00:00:00
2014-07-10 00:00:00
2014-07-11 00:00:00
2014-07-12 00:00:00
2014-07-13 00:00:00
2014-07-14 00:00:00
2014-07-15 00:00:00
2014-07-16 00:00:00
2014-07-17 00:00:00
2014-07-18 00:00:00
2014-07-19 00:00:00
2014-07-20 00:00:00
2014-07-21 00:00:00
2014-07-22 00:00:00
2014-07-23 00:00:00
2014-07-24 00:00:00
2014-07-25 00:00:00
2014-07-26 00:00:00
2014-07-27 00:00:00
2014-07-28 00:00:00
2014-07-29 00:00:00
2014-07-30 00:00:00
2014-07-31 00:00:00
2014-08-01 00:00:00
2014-08-02 00:00:00
2014-08-03 00:00:00
2014-08-04 00:00:00
2014-08-05 00:00:00
2014-08-06 00:00:00
2014-08-07 00:00:00
2014-08-08 00:00:00
2014-08-09 00:00:00
2014-08-10 00:00:00
2014-08-11 00:00:00
2014-08-12 00:00:00
2014-08-13 00:00:00
2014-08-14 00:00:00
2014-08-15 00:00:00
2014-08-16 00:00:00
2014-08-17 00:00:00
2014-08-18 00:00:00
2014-08-19 00:00:00
2014-08-20 00:00:00
2014-08-21 00:00:00
2014-08-22 00:00:00
2014-08-23 00:00:00
2014-08-24 00:00:00
2014-08-25 00:00:00
2014-08-26 00:00:00
2014-08-27 00:00:00
2014-08-28 00:00:00
2014-08-29 00:00:00
2014-08-30 00:00:00
2014-08-31 00:00:00
2014-09-01 00:00:00
2014-09-02 00:00:00
2014-09-03 00:00:00
2014-09-04 00:00:00
2014-09-05 00:00:00
2014-09-06 00:00:00
2014-09-07 00:00:00
2014-09-08 00:00:00
2014-09-09 00:00:00
2014-09-10 00:00:00
2014-09-11 00:00:00
2014-09-12 00:00:00
2014-09-13 00:00:00
2014-09-14 00:00:00
2014-09-15 00:00:00
2014-09-16 00:00:00
2014-09-17 00:00:00
2014-09-18 00:00:00
2014-09-19 00:00:00
2014-09-20 00:00:00
2014-09-21 00:00:00
2014-09-22 00:00:00
2014-09-23 00:00:00
2014-09-24 00:00:00
2014-09-25 00:00:00
2014-09-26 00:00:00
2014-09-27 00:00:00
2014-09-28 00:00:00
2014-09-29 00:00:00
2014-09-30 00:00:00
2014-10-01 00:00:00
2014-10-02 00:00:00
2014-10-03 00:00:00
2014-10-04 00:00:00
2014-10-05 00:00:00
2014-10-06 00:00:00
2014-10-07 00:00:00
2014-10-08 00:00:00
2014-10-09 00:00:00
2014-10-10 00:00:00
2014-10-11 00:00:00
2014-10-12 00:00:00
2014-10-13 00:00:00
2014-10-14 00:00:00
2014-10-15 00:00:00
2014-10-16 00:00:00
2014-10-17 00:00:00
2014-10-18 00:00:00
2014-10-19 00:00:00
2014-10-20 00:00:00
2014-10-21 00:00:00
2014-10-22 00:00:00
2014-10-23 00:00:00
2014-10-24 00:00:00
2014-10-25 00:00:00
2014-10-26 00:00:00
2014-10-27 00:00:00
2014-10-28 00:00:00
2014-10-29 00:00:00
2014-10-30 00:00:00
2014-10-31 00:00:00
2014-11-01 00:00:00
2014-11-02 00:00:00
2014-11-03 00:00:00
2014-11-04 00:00:00
2014-11-05 00:00:00
2014-11-06 00:00:00
2014-11-07 00:00:00
2014-11-08 00:00:00
2014-11-09 00:00:00
2014-11-10 00:00:00
2014-11-11 00:00:00
2014-11-12 00:00:00
2014-11-13 00:00:00
2014-11-14 00:00:00
2014-11-15 00:00:00
2014-11-16 00:00:00
2014-11-17 00:00:00
2014-11-18 00:00:00
2014-11-19 00:00:00
2014-11-20 00:00:00
2014-11-21 00:00:00
2014-11-22 00:00:00
2014-11-23 00:00:00
2014-11-24 00:00:00
2014-11-25 00:00:00
2014-11-26 00:00:00
2014-11-27 00:00:00
2014-11-28 00:00:00
2014-11-29 00:00:00
2014-11-30 00:00:00
2014-12-01 00:00:00
2014-12-02 00:00:00
2014-06-20 00:00:00
2014-06-21 00:00:00
2014-06-22 00:00:00
2014-06-23 00:00:00
2014-06-24 00:00:00
2014-06-25 00:00:00
2014-06-26 00:00:00
2014-06-27 00:00:00
2014-06-28 00:00:00
2014-06-29 00:00:00
2014-06-30 00:00:00
2014-07-01 00:00:00
2014-07-02 00:00:00
2014-07-03 00:00:00
2014-07-04 00:00:00
2014-07-05 00:00:00
2014-07-06 00:00:00
2014-07-07 00:00:00
2014-07-08 00:00:00
2014-07-09 00:00:00
2014-07-10 00:00:00
2014-07-11 00:00:00
2014-07-12 00:00:00
2014-07-13 00:00:00
2014-07-14 00:00:00
2014-07-15 00:00:00
2014-07-16 00:00:00
2014-07-17 00:00:00
2014-07-18 00:00:00
2014-07-19 00:00:00
2014-07-20 00:00:00
2014-07-21 00:00:00
2014-07-22 00:00:00
2014-07-23 00:00:00
2014-07-24 00:00:00
2014-07-25 00:00:00
2014-07-26 00:00:00
2014-07-27 00:00:00
2014-07-28 00:00:00
2014-07-29 00:00:00
2014-07-30 00:00:00
2014-07-31 00:00:00
2014-08-01 00:00:00
2014-08-02 00:00:00
2014-08-03 00:00:00
2014-08-04 00:00:00
2014-08-05 00:00:00
2014-08-06 00:00:00
2014-08-07 00:00:00
2014-08-08 00:00:00
2014-08-09 00:00:00
2014-08-10 00:00:00
2014-08-11 00:00:00
2014-08-12 00:00:00
2014-08-13 00:00:00
2014-08-14 00:00:00
2014-08-15 00:00:00
2014-08-16 00:00:00
2014-08-17 00:00:00
2014-08-18 00:00:00
2014-08-19 00:00:00
2014-08-20 00:00:00
2014-08-21 00:00:00
2014-08-22 00:00:00
2014-08-23 00:00:00
2014-08-24 00:00:00
2014-08-25 00:00:00
2014-08-26 00:00:00
2014-08-27 00:00:00
2014-08-28 00:00:00
2014-08-29 00:00:00
2014-08-30 00:00:00
2014-08-31 00:00:00
2014-09-01 00:00:00
2014-09-02 00:00:00
2014-09-03 00:00:00
2014-09-04 00:00:00
2014-09-05 00:00:00
2014-09-06 00:00:00
2014-09-07 00:00:00
2014-09-08 00:00:00
2014-09-09 00:00:00
2014-09-10 00:00:00
2014-09-11 00:00:00
2014-09-12 00:00:00
2014-09-13 00:00:00
2014-09-14 00:00:00
2014-09-15 00:00:00
2014-09-16 00:00:00
2014-09-17 00:00:00
2014-09-18 00:00:00
2014-09-19 00:00:00
2014-09-20 00:00:00
2014-09-21 00:00:00
2014-09-22 00:00:00
2014-09-23 00:00:00
2014-09-24 00:00:00
2014-09-25 00:00:00
2014-09-26 00:00:00
2014-09-27 00:00:00
2014-09-28 00:00:00
2014-09-29 00:00:00
2014-09-30 00:00:00
2014-10-01 00:00:00
2014-10-02 00:00:00
2014-10-03 00:00:00
2014-10-04 00:00:00
2014-10-05 00:00:00
2014-10-06 00:00:00
2014-10-07 00:00:00
2014-10-08 00:00:00
2014-10-09 00:00:00
2014-10-10 00:00:00
2014-10-11 00:00:00
2014-10-12 00:00:00
2014-10-13 00:00:00
2014-10-14 00:00:00
2014-10-15 00:00:00
2014-10-16 00:00:00
2014-10-17 00:00:00
2014-10-18 00:00:00
2014-10-19 00:00:00
2014-10-20 00:00:00
2014-10-21 00:00:00
2014-10-22 00:00:00
2014-10-23 00:00:00
2014-10-24 00:00:00
2014-10-25 00:00:00
2014-10-26 00:00:00
2014-10-27 00:00:00
2014-10-28 00:00:00
2014-10-29 00:00:00
2014-10-30 00:00:00
2014-10-31 00:00:00
2014-11-01 00:00:00
2014-11-02 00:00:00
2014-11-03 00:00:00
2014-11-04 00:00:00
2014-11-05 00:00:00
2014-11-06 00:00:00
2014-11-07 00:00:00
2014-11-08 00:00:00
2014-11-09 00:00:00
2014-11-10 00:00:00
2014-11-11 00:00:00
2014-11-12 00:00:00
2014-11-13 00:00:00
2014-11-14 00:00:00
2014-11-15 00:00:00
2014-11-16 00:00:00
2014-11-17 00:00:00
2014-11-18 00:00:00
2014-11-19 00:00:00
2014-11-20 00:00:00
2014-11-21 00:00:00
2014-11-22 00:00:00
2014-11-23 00:00:00
2014-11-24 00:00:00
2014-11-25 00:00:00
2014-11-26 00:00:00
2014-11-27 00:00:00
2014-11-28 00:00:00
2014-11-29 00:00:00
2014-11-30 00:00:00
2014-12-01 00:00:00
2014-12-02 00:00:00
2014-06-20 00:00:00
2014-06-21 00:00:00
2014-06-22 00:00:00
2014-06-23 00:00:00
2014-06-24 00:00:00
2014-06-25 00:00:00
2014-06-26 00:00:00
2014-06-27 00:00:00
2014-06-28 00:00:00
2014-06-29 00:00:00
2014-06-30 00:00:00
2014-07-01 00:00:00
2014-07-02 00:00:00
2014-07-03 00:00:00
2014-07-04 00:00:00
2014-07-05 00:00:00
2014-07-06 00:00:00
2014-07-07 00:00:00
2014-07-08 00:00:00
2014-07-09 00:00:00
2014-07-10 00:00:00
2014-07-11 00:00:00
2014-07-12 00:00:00
2014-07-13 00:00:00
2014-07-14 00:00:00
2014-07-15 00:00:00
2014-07-16 00:00:00
2014-07-17 00:00:00
2014-07-18 00:00:00
2014-07-19 00:00:00
2014-07-20 00:00:00
2014-07-21 00:00:00
2014-07-22 00:00:00
2014-07-23 00:00:00
2014-07-24 00:00:00
2014-07-25 00:00:00
2014-07-26 00:00:00
2014-07-27 00:00:00
2014-07-28 00:00:00
2014-07-29 00:00:00
2014-07-30 00:00:00
2014-07-31 00:00:00
2014-08-01 00:00:00
2014-08-02 00:00:00
2014-08-03 00:00:00
2014-08-04 00:00:00
2014-08-05 00:00:00
2014-08-06 00:00:00
2014-08-07 00:00:00
2014-08-08 00:00:00
2014-08-09 00:00:00
2014-08-10 00:00:00
2014-08-11 00:00:00
2014-08-12 00:00:00
2014-08-13 00:00:00
2014-08-14 00:00:00
2014-08-15 00:00:00
2014-08-16 00:00:00
2014-08-17 00:00:00
2014-08-18 00:00:00
2014-08-19 00:00:00
2014-08-20 00:00:00
2014-08-21 00:00:00
2014-08-22 00:00:00
2014-08-23 00:00:00
2014-08-24 00:00:00
2014-08-25 00:00:00
2014-08-26 00:00:00
2014-08-27 00:00:00
2014-08-28 00:00:00
2014-08-29 00:00:00
2014-08-30 00:00:00
2014-08-31 00:00:00
2014-09-01 00:00:00
2014-09-02 00:00:00
2014-09-03 00:00:00
2014-09-04 00:00:00
2014-09-05 00:00:00
2014-09-06 00:00:00
2014-09-07 00:00:00
2014-09-08 00:00:00
2014-09-09 00:00:00
2014-09-10 00:00:00
2014-09-11 00:00:00
2014-09-12 00:00:00
2014-09-13 00:00:00
2014-09-14 00:00:00
2014-09-15 00:00:00
2014-09-16 00:00:00
2014-09-17 00:00:00
2014-09-18 00:00:00
2014-09-19 00:00:00
2014-09-20 00:00:00
2014-09-21 00:00:00
2014-09-22 00:00:00
2014-09-23 00:00:00
2014-09-24 00:00:00
2014-09-25 00:00:00
2014-09-26 00:00:00
2014-09-27 00:00:00
2014-09-28 00:00:00
2014-09-29 00:00:00
2014-09-30 00:00:00
2014-10-01 00:00:00
2014-10-02 00:00:00
2014-10-03 00:00:00
2014-10-04 00:00:00
2014-10-05 00:00:00
2014-10-06 00:00:00
2014-10-07 00:00:00
2014-10-08 00:00:00
2014-10-09 00:00:00
2014-10-10 00:00:00
2014-10-11 00:00:00
2014-10-12 00:00:00
2014-10-13 00:00:00
2014-10-14 00:00:00
2014-10-15 00:00:00
2014-10-16 00:00:00
2014-10-17 00:00:00
2014-10-18 00:00:00
2014-10-19 00:00:00
2014-10-20 00:00:00
2014-10-21 00:00:00
2014-10-22 00:00:00
2014-10-23 00:00:00
2014-10-24 00:00:00
2014-10-25 00:00:00
2014-10-26 00:00:00
2014-10-27 00:00:00
2014-10-28 00:00:00
2014-10-29 00:00:00
2014-10-30 00:00:00
2014-10-31 00:00:00
2014-11-01 00:00:00
2014-11-02 00:00:00
2014-11-03 00:00:00
2014-11-04 00:00:00
2014-11-05 00:00:00
2014-11-06 00:00:00
2014-11-07 00:00:00
2014-11-08 00:00:00
2014-11-09 00:00:00
2014-11-10 00:00:00
2014-11-11 00:00:00
2014-11-12 00:00:00
2014-11-13 00:00:00
2014-11-14 00:00:00
2014-11-15 00:00:00
2014-11-16 00:00:00
2014-11-17 00:00:00
2014-11-18 00:00:00
2014-11-19 00:00:00
2014-11-20 00:00:00
2014-11-21 00:00:00
2014-11-22 00:00:00
2014-11-23 00:00:00
2014-11-24 00:00:00
2014-11-25 00:00:00
2014-11-26 00:00:00
2014-11-27 00:00:00
2014-11-28 00:00:00
2014-11-29 00:00:00
2014-11-30 00:00:00
2014-12-01 00:00:00
2014-12-02 00:00:00
```

And if i not use fill(0)

```
>>> data = client.query("select count(timespan) as visits from tracking_file where sharer_id = '34f72b47-847c-4a12-a64e-43f459d6e9aa' and file_id = '673b65b4-0866-4020-a391-58697d0e2422'  and time > 1403253505s and time < 1417514972s group by time(24h) order asc")
>>> data
[{u'points': [[1403222400, 1], [1405123200, 2], [1416182400, 2]], u'name': u'tracking_file', u'columns': [u'time', u'visits']}]
>>> for p in data[0]['points']:
...     print datetime.fromtimestamp(int(p[0]))
... 
2014-06-20 00:00:00
2014-07-12 00:00:00
2014-11-17 00:00:00

```
all correct, what problem with fill(0)

p/s: i not use shard data...

**Comment by somebody32** [comment]
The same problem with `fill(0)`, `group by time(1d)`

**Comment by jjmalina** [comment]
I'm having this issue without even grouping by time. I write data pre-aggregated by hour and use sequence numbers (by taking the hash of the iso8601 timestamp and series name).

**Comment by maxdec** [comment]
+1, I also get this error with `fill(0)` + `group by time(1d)`. I'm using v0.8.8. A friend doesn't seem affected with v0.8.6.

**Comment by pauldix** [comment]
This is fixed in 0.9.0. We won't be making additional fixes to the 0.8 line.
